app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -141,6 +141,30 @@ def bulk_cancel
         }
       end
 
+      api :POST, '/tasks/bulk_stop', N_('Stop all stoppable tasks')
+      param :search, String, :desc => N_('Stop tasks matching search string')
+      param :task_ids, Array, :desc => N_('Stop specific tasks by ID')
+      def bulk_stop
+        if params[:search].nil? && params[:task_ids].nil?
+          raise BadRequest, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+
+        filtered_scope = bulk_scope
+        to_stop = filtered_scope.where.not(state: :stopped)
+        already_stopped_length = filtered_scope.length - to_stop.length
```

Comment:
Using `length` would load all the records into memory and then count them. On the other hand, using `count` would make the database count the records and return their number, which is more efficient.

Reply:
using now length for `to_stop` as its loaded anyway and count for `filtered_scope`